spatialos-sdk/src/worker/op.rs

```diff
@@ -455,7 +455,23 @@ pub struct EntityQueryResponseOp {
 #[derive(Debug)]
 pub struct AddComponentOp {
     pub entity_id: EntityId,
-    pub component_data: component::internal::ComponentData,
+    component_data: component::internal::ComponentData,
+}
+
+impl AddComponentOp {
+    pub fn component_id(&self) -> ComponentId {
+        self.component_data.component_id
+    }
+
+    pub fn get<C: Component>(&self) -> Option<&C> {
+        if C::ID == self.component_id() {
```

Comment:
This is still unsafe if the user opts to do manual serialization (which admittedly is not really possible yet), because `user_handle` will be `null`. However, I am happy to merge this now and implement that case later.

Reply:
Ah, that's a good point. I'll add a null check to the pointer and a panic noting that manual serialization isn't implemented yet 👍